Sources/BuildSystemIntegration/BuildSystemManager.swift

```diff
@@ -327,11 +334,9 @@ package actor BuildSystemManager: QueueBasedMessageHandler {
 
   private var cachedTargetSources = RequestCache<BuildTargetSourcesRequest>()
 
-  /// The parameters with which `SourceFilesAndDirectories` can be cached in `cachedSourceFilesAndDirectories`.
-  private struct SourceFilesAndDirectoriesKey: Hashable {
-    let includeNonBuildableFiles: Bool
-    let sourcesItems: [SourcesItem]
-  }
+  /// `SourceFilesAndDirectories` is a global property that only gets reset when the build targets change and thus
+  /// has no real key.
```

Comment:
Is it being a cache still useful in that case?

Reply:
Constructing `SourceFilesAndDirectories` is still non-trivial and caching it has significant performance benefits because we call it during `canonicalTarget(for:)`, which gets call pretty frequently.

Reply:
Sorry, I didn't mean to not "cache" it. I just meant is there any point using `Cache` rather than just having a `SourceFilesAndDirectories` directly.

Reply:
We don’t need to but I prefer to because it aligns this property with all the other cached properties and allow us to use the `get` function that computes on-demand, stores it in a `Task` etc. So, there’s still enough of a benefit from using `Cache`.

Reply:
Okay 👍